chips/e310x/src/chip.rs

```diff
@@ -52,7 +52,7 @@ impl kernel::Chip for E310x {
             while let Some(interrupt) = plic::next_pending() {
                 match interrupt {
                     interrupts::UART0 => uart::UART0.handle_interrupt(),
-                    index @ interrupts::GPIO0..interrupts::GPIO31 => {
+                    index @ interrupts::GPIO0..=interrupts::GPIO31 => {
                         gpio::PORT[index as usize].handle_interrupt()
```

Comment:
The removal of the `exclusive_range_pattern` feature looks good, although there is (I believe) a bug remaining in the code:
`interrupts::GPIO0` has a value of 8, so using the raw range to index into `gpio::PORT` will be off by 8 and overflow for the higher pins.  I meant to circle back to this after [fixing it in Ibex](https://github.com/tock/tock/blob/d68ec06d4b01aeb8e3cfb91c02e5215525a9d567/chips/ibex/src/chip.rs#L46)